### PR TITLE
Long standing: add support for templates and other minor changes.

### PR DESCRIPTION
**Pull request by asheshambasta**
This PR is probably bitrotten, but it suffices to review it nonetheless. 

Since this is a third party library, some of our code standards don't apply.

### CODE REVIEW

File: src/Network/API/Mandrill/Templates/Types.hs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import qualified Data.Text                  as T
+import           Lens.Micro.TH              (makeLenses)
+
+import           Network.API.Mandrill.Types
+
+-- | Request to get templates
+data GetTemplatesRq = GetTemplatesRq
+    { _gtrq_key   :: MandrillKey -- ^ API Key
+    -- ^ An optional Template label (for searching)
+    , _gtrq_label :: Maybe Label -- ^ An optional Template label (for searching)
```
Comment (fpringle) [inline src/Network/API/Mandrill/Templates/Types.hs:34]: CamelCase?

File: src/.dir-locals.el
Comment (fpringle) [inline src/.dir-locals.el:1]: Can we get rid of this and other `.dir-locals.el` files?

**Review by fpringle — CHANGES_REQUESTED** [review]
LGTM, small nitpicking.

File: src/Network/API/Mandrill/Templates/Types.hs (hunk 1 above)
Comment (asheshambasta) [reply src/Network/API/Mandrill/Templates/Types.hs:34]: Its a fork, so I kept it in the same style as before. Maybe we should just do a full refactor someday.

File: src/.dir-locals.el
Comment (asheshambasta) [reply src/.dir-locals.el:1]: Done.

**Review by fpringle — APPROVED** [review]

**Pull request merged**